src/components/Countdown.js

```diff
@@ -1,49 +1,67 @@
 import React, { useEffect, useState } from "react";
+import {
+  formatDays,
+  formatHours,
+  formatMinute,
+  formatSeconds,
+} from "../utils/utils";
 
-const calculateTimeLeft = () => {
-  let year = new Date().getFullYear();
-  // const endTime = new Date(year, 1, 21, 9, 30, 0, 0); // Feb 21, 9:30am
-  const endTime = new Date(year, 1, 19, 18, 0, 0, 0); // Feb 19, 6:00pm
-  const difference = +endTime - +new Date();
+const calculateTimeLeft = (difference) => {
   let timeLeft = {};
+  let year = new Date().getFullYear();
+  const endTime = new Date(year, 1, 21, 9, 30, 0, 0); // Feb 21, 9:30am
+  const startTime = new Date(year, 1, 19, 18, 0, 0, 0); // Feb 19, 6:00pm
+  difference["startDifference"] = +startTime - +new Date();
+  difference["endDifference"] = +endTime - +new Date();
 
-  if (difference > 0) {
-    timeLeft = {
-      days: Math.floor(difference / (1000 * 60 * 60 * 24)),
-      hours: String(Math.floor((difference / (1000 * 60 * 60)) % 24)).padStart(
-        2,
-        "0"
-      ),
-      minutes: String(Math.floor((difference / 1000 / 60) % 60)).padStart(
-        2,
-        "0"
-      ),
-      seconds: String(Math.floor((difference / 1000) % 60)).padStart(2, "0"),
-    };
+  let diff, days, hours, minutes, seconds;
+  if (difference.startDifference > 0) {
+    diff = difference.startDifference;
+  } else if (difference.endDifference > 0) {
+    diff = difference.endDifference;
   }
 
+  days = formatDays(diff);
+  hours = formatHours(diff);
+  minutes = formatMinute(diff);
+  seconds = formatSeconds(diff);
+
+  timeLeft = {
+    days: days,
+    hours: hours,
+    minutes: minutes,
+    seconds: seconds,
+  };
+
   return timeLeft;
 };
 
-const useCounter = () => {
-  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft());
+const useCounter = (difference) => {
+  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft(difference));
 
   useEffect(() => {
-    const interval = setInterval(() => setTimeLeft(calculateTimeLeft()), 1000);
+    const interval = setInterval(
+      () => setTimeLeft(calculateTimeLeft(difference)),
+      1000
+    );
     return () => clearInterval(interval);
-  }, []);
+  }, [difference]);
```

Comment:
good to note that this doesn't re-run all the time, because difference is object reference (doesn't get reset)